acinclude/nettle.m4

```diff
@@ -15,8 +15,8 @@ AC_DEFUN([SQUID_CHECK_NETTLE_BASE64],[
   AH_TEMPLATE(HAVE_NETTLE34_BASE64,[set to 1 if Nettle 3.4 API will link])
   SQUID_STATE_SAVE(squid_nettle_base64_state)
   AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+#   include <cinttypes>
 #   include <cstddef>
-#   include <cstdint>
```

Comment:
> PR description: With current C/C++ standard the intypes.h includes stdint.h

Agreed. Due to the above guarantee, we do not need to include stdint.h if we have to include intypes.h.

> PR title (and this code change): Include inttypes instead of stdint

Why? The two headers are different. If some code only needs (smaller) stdint.h, why should that code include (larger) inttypes.h that it does not need?

Please note that I am not saying that this particular AC_LANG_PROGRAM() or any other specific place changed in this PR does not need inttypes.h. I suspect some of them needed that header, triggering this PR; I have not checked any specific locations for that code needs. It is the "let's always include the larger header" approach (implied by the PR title/description) that raises a red flag, even if some code does need to include the larger header. If all code changes in this PR are required, then let's adjust this PR title/description to remove that implication. For example:

```
Fix ... build by including missing inttypes.h

Also, standard C/C++ intypes.h includes stdint.h, so we do not need to
explicitly include the latter when we need to include the former.
```

Reply:
This line is being touched ensure that the configure test is checking things work with the same includes that Squid itself uses later for the actual build.

> > PR description: With current C/C++ standard the intypes.h includes stdint.h
> 
> Agreed. Due to the above guarantee, we do not need to include stdint.h if we have to include intypes.h.
> 
> > PR title (and this code change): Include inttypes instead of stdint
> 
> Why? The two headers are different. If some code only needs (smaller) stdint.h, why should that code include (larger) inttypes.h that it does not need?
> 

Why? because these files are globally included the ```cstdint``` and ```stdint.h``` is redundant with C99 and C++11.

> Please note that I am not saying that this particular AC_LANG_PROGRAM() or any other specific place changed in this PR does not need inttypes.h. I suspect some of them needed that header, triggering this PR; I have not checked any specific locations for that code needs.

FTR, the trigger for this PR was your request in other reviews to use the STD definitions for printf ```%PRI...``` codes. That led to reviewing the ```compat/types.h``` content against C++11 guaranteed available symbols.

This is basic code maintenance.

Reply:
I am resolving this change request and using the information you have provided in more specific change requests.